tests/integ/modin/frame/test_quantile.py

```diff
@@ -36,16 +37,34 @@
 def test_quantile_basic(q, interpolation):
     snow_df = pd.DataFrame(TEST_QUANTILE_DATA)
     native_df = native_pd.DataFrame(TEST_QUANTILE_DATA)
-    expected_query_count = 2 if isinstance(q, list) else 0
-
-    with SqlCounter(query_count=1, union_count=expected_query_count):
+    with SqlCounter(
+        query_count=1, union_count=0 if isinstance(q, float) else len(q) - 1
+    ):
         eval_snowpark_pandas_result(
             snow_df,
             native_df,
             lambda df: df.quantile(q, numeric_only=True),
         )
 
 
+@pytest.mark.parametrize("q", TEST_QUANTILES)
+def test_quantile_single_column(q):
+    # When the input frame has only a single column, and q is sorted, no UNION ALLs are performed
+    expected_union_count = len(q) - 1 if sorted(q) != q else 0
+    snow_df, native_df = create_test_dfs({"a": [1, 2, 3, 4]})
```

Comment:
nit: you can parametrize over data to save some similar code here, e.g.

```
@pytest.mark.parametrize("data", [{"a": [1, 2, 3, 4]}, {"a": [1, 2, 3, 4], "b": ["the", "quick", "brown", "fox"]}])
```

Reply:
The operation is different between the two (the 2-column frame uses `numeric_only=True`), and I don't think it's worth doing a double parametrization for just 2 function calls.